### PR TITLE
Don't clear sessions when clearing cache

### PR DESCRIPTION
**Pull request by MegaphoneJon**
Overview
----------------------------------------
Clearing cache shouldn't break donations in progress, but it does.

Before
----------------------------------------
See above.

After
----------------------------------------
Nope

Comments
----------------------------------------
Original patch from @mlutfy with modifications to clear most, but not all, of `civicrm_cache`.

### CODE REVIEW

File: CRM/Core/Config.php [hunk 1]
```diff
@@ -287,7 +287,9 @@ public function cleanupCaches($sessionReset = TRUE) {
     UserJob::delete(FALSE)->addWhere('expires_date', '<', 'now')->execute();
     // clear all caches
     self::clearDBCache();
-    Civi::cache('session')->clear();
+    // [ML] SYMBIOTIC This causes problems if people are filling-in quickform
+    // forms during a system.flush
+    // Civi::cache('session')->clear();
```
Comment (totten) [inline CRM/Core/Config.php:292]: In this context, it receives the param `$sessionReset`. Notwithstanding the method-signature here, the actual/typical default should `FALSE`. (For purposes of `cv flush`, `System.flush`, `civicrm/menu/rebuild`, etc -- they all call via `rebuildMenuAndCaches(...$sessionReset=FALSE)`). It seems better to abide that flag, no?

**Comment by agileware-justin** [comment]
Thanks @MegaphoneJon 👏🏻

**Comment by mattwire** [comment]
At the Montreal sprint @MegaphoneJon and I agreed that we should get this into core. Quite a few of us have been running versions of this on client sites for quite a long time.

File: CRM/Core/Config.php (hunk 1 above)
Comment (mlutfy) [reply CRM/Core/Config.php:292]: if we change the param `$sessionReset = FALSE` by default, then add an "if" around ` Civi::cache('session')->clear()`, then yes, it works as intended I think.

Tested with:

- `cv flush`
- "flush cache" from the admin UI
- `cv api system.flush session=1` => will flush the QF cache (sessionReset=true)

**Comment by mlutfy** [comment]
This would be great. Thank you for pushing this forward.

I did a bit more testing based on Tim's comment, and I think it would be sensible to:

- default `sessionReset = FALSE`  in the function params
- add an "if" around the session flush line, instead of removing it

We can also strip most the comments out of the patch (ex: "ML SYMBIOTIC").

**Comment by mattwire** [comment]
Thanks @bgm. @MegaphoneJon what do you think of the updates?

**Pull request merged**